torch/_dynamo/variables/builder.py

```diff
@@ -1686,24 +1686,36 @@ def wrap_module(self, value: torch.nn.Module):
             self.install_guards(GuardBuilder.TYPE_MATCH)
             if torch._dynamo.config.inline_inbuilt_nn_modules:
                 freezing = is_parameter_freezing()
-                # Guard against the case where user may overwrite named parameters
-                # / named buffers
-                # NOTE: This is not likely to happen but worth guarding to avoid
-                # exception
-                if (
-                    callable(value.named_parameters)
-                    and value.named_parameters.__func__
-                    is og_module_named_parameters_fn_ptr
-                ):
-                    for _, p in value.named_parameters():
-                        self.mark_static_input(p, guard=freezing)
-
-                if (
-                    callable(value.named_buffers)
-                    and value.named_buffers.__func__ is og_module_named_buffers_fn_ptr
-                ):
-                    for _, b in value.named_buffers():
-                        self.mark_static_input(b, guard=freezing)
+
+                try:  # catch TypeErrors from unserializable nn modules
+                    # Guard against the case where user may overwrite named parameters
+                    # / named buffers
+                    # NOTE: This is not likely to happen but worth guarding to avoid
+                    # exception
+                    if (
+                        callable(value.named_parameters)
+                        and value.named_parameters.__func__
+                        is og_module_named_parameters_fn_ptr
+                    ):
+                        for _, p in value.named_parameters():
+                            self.mark_static_input(p, guard=freezing)
+
+                    if (
+                        callable(value.named_buffers)
+                        and value.named_buffers.__func__
+                        is og_module_named_buffers_fn_ptr
+                    ):
+                        for _, b in value.named_buffers():
+                            self.mark_static_input(b, guard=freezing)
+
+                    if freezing:
+                        # we need to add the module to tracing context
+                        # in order to allow its params to get invalidated
+                        # this will get cleaned up once compile ends
+                        self.tx.output.nn_modules[self.name] = value
+
+                except TypeError as e:
+                    raise ObservedException(e)
```

Comment:
Call raise_observed_exception here - that's the API for creating them :). (Rather than doing `raise ObservedException).

Reply:
Fixed.